### DISCUSSION TITLE
updateRpcUrl() is not a function in Ethereum Provider

### DISCUSSION THREAD

**Discussion by tiennt0212** [WalletConnect/walletconnect-monorepo#2688] [Developer Support] [answered]
### My application with WalletConnect 1.0
I've used `@walletconnect/web3-provider` before.
In my application, I have some special logic and I use the updateRpcUrl() to update the `rpcUrl` of a specific `chainId`.

### Migrating to the WalletConnect 2.0
I was recommended to use `@walletconnect/ethereum-provider`.
It looks okay for now, however, I can't util the updateRpcUrl() method to supply my logical handler.

**Help me:** Does Ethereum Provider has any method to update `rpcUrl` of a specific `chainId`?

**Comment by glitch-txs** [comment, marked as answer]
you can try this:

```ts
const chainId: number = 1 //for mainnet 
const rpcUrl: string = "..."
provider.signer.setDefaultChain(`eip155:${chainId}`, rpcUrl);
```

**Reply by tiennt0212** [reply]
It worked, thank you :sunflower:
